src/forwarder.rs

```diff
@@ -153,25 +157,54 @@ fn recv_from_channel_and_send_multiple_dest(
         |_received_packet, _is_already_marked_as_discard, _is_dup| {},
     );
 
+    let (trace_shreds, packets): (Vec<TraceShred>, Vec<Packet>) = packet_batch_vec[0]
```

Comment:
kinda concerned with potential perf hit of doing this, wonder if it makes more sense as a normal shred with some special data encoded in it? need to look at BE PR still

Reply:
the trace shred is 24 bytes in my measurements

Reply:
mainly talking about the cloned, decode attempt, and any other logic here

Reply:
we can extract this to another stage (eg shred_verify) if you'd prefer

Reply:
maybe we can add a cli flag to print tracer stats?

Reply:
so its like a debug mode

Reply:
do you want this to be turned on/off by that cli flag? https://github.com/jito-labs/shredstream-proxy/blob/d703b65682b8f2ffebf80a516e46db4511d19849/src/forwarder.rs#L201

Reply:
right now it is turned on if we have SOLANA_CLUSTER and REGION env vars.

Reply:
@buffalu we can just forward the trace packet to validator, and let them discard it since it wont pass sigverify. any thoughts?

Reply:
another option is to just mark them trace pkt as discard too

Reply:
hmm yeah that works too. imo we add a flag for this like debug-shred-trace or whatever